modules/programs/obsidian.nix

```diff
@@ -0,0 +1,547 @@
+                          modifiers = mkOption {
+                            type = with types; listOf str;
+                            description = "The hotkey modifiers.";
+                            default = [ ];
+                          };
+
+                          key = mkOption {
+                            type = types.str;
+                            description = "The hotkey.";
+                          };
+                        };
+                      };
+                  in
+                  mkOption {
+                    description = "Hotkeys to configure.";
+                    type = with types; attrsOf (listOf (submodule hotkeysOptions));
+                    default = cfg.defaultSettings.hotkeys;
+                  };
+
+                extraFiles =
+                  let
+                    extraFilesOptions =
+                      { name, config, ... }:
+                      {
+                        options = {
+                          source = mkOption {
+                            type = with types; nullOr path;
+                            description = "Path of the source file or directory.";
+                            default = null;
+                          };
+
+                          text = mkOption {
+                            type = with types; nullOr str;
+                            description = "Text of the file.";
+                            default = null;
+                          };
+
+                          target = mkOption {
+                            type = types.str;
+                            defaultText = literalExpression "name";
+                            description = "Path to target relative to the vault's directory.";
+                          };
+                        };
+
+                        config.target = mkDefault name;
+                      };
+                  in
+                  mkOption {
+                    description = "Extra files to link to the vault directory.";
+                    type = with types; attrsOf (submodule extraFilesOptions);
+                    default = cfg.defaultSettings.extraFiles;
+                  };
+              };
+            };
+
+            config.target = mkDefault name;
+          }
+        )
+      );
+      default = { };
+    };
+  };
+
+  config =
+    let
+      vaults = builtins.filter (vault: vault.enable == true) (builtins.attrValues cfg.vaults);
+      getManifest =
+        item:
+        let
+          manifest = builtins.fromJSON (builtins.readFile "${item.pkg}/manifest.json");
+        in
+        manifest.id or manifest.name;
+    in
+    lib.mkIf cfg.enable {
+      home = {
+        packages = [ cfg.package ];
+
+        file =
+          let
+            mkApp = vault: {
+              name = "${vault.target}/.obsidian/app.json";
+              value.source = (pkgs.formats.json { }).generate "app.json" vault.settings.app;
+            };
```

Comment:
I worry this will feel unintuitive. If someone declares `defaultSettings.app` and then `vault.settings.app` the only thing that gets generated is `vault.settings.app`.